client/python/tests/test_facet.py

```diff
@@ -199,6 +201,65 @@ def test_ownership_job_facet():
     assert expected_event == event_sent
 
 
+def test_tag_dataset_facet():
+    session = MagicMock()
+    client = OpenLineageClient(url="http://example.com", session=session)
+
+    tag_dataset_facet = {
+        "tags": [{
+            "key": "pii",
+            "value": "true",
+            "field": "email"
+        }, {
+            "key": "pii",
+            "value": "true",
+            "field": "name"
+        }, {
+            "key": "public",
+            "value": "false"
+        }],
+        "_producer": "https://github.com/OpenLineage/OpenLineage/tree/0.0.1/client/python",
+        "_schemaURL": "https://raw.githubusercontent.com/OpenLineage/OpenLineage/main/spec"
+                      "/OpenLineage.json#/definitions/TagDatasetFacet",
+    }
```

Comment:
Should we have a schema similar to the column lineage facet for consistency? also key-value tags sound like a good match for additional fields?
https://github.com/OpenLineage/OpenLineage/blob/main/spec/facets/ColumnLineageDatasetFacet.md
```
"tags": {
  "dataset": { 
    "public": "false"
  },
  "fields": {
    "email": {
      "pii": "true"
    },
    "name": {
      "pii": "true"
    }
  }
}
```

Reply:
Having separate `dataset` and `fields` objects makes sense - I feel it's more obvious what's happening.

Regarding additional fields - I've usually found they are quite awkward to use, but it's not a strong opinion. If you or someone else prefers it that way I can agree with it.

Reply:
> If you or someone else prefers it that way I can agree with it.

I've put same comment in the discussion above https://github.com/OpenLineage/OpenLineage/pull/1630#discussion_r1105847657

Reply:
I do think it is easier to read that way.